demos/browser/app/meetingV2/video/PaginationManager.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+export default class PaginationManager<Type> {
+  private currentPageStart: number = 0;
+
+  private all = new Array<Type>();
+  
+  constructor(private pageSize: number) {}
+
+  currentPage(): Array<Type> {
+    return this.all.slice(this.currentPageStart, this.currentPageStart + this.pageSize);
+  }
+
+  add(toAdd: Type) {
+    if (this.all.includes(toAdd)) {
+      return;
+    }
+    this.all.push(toAdd);
+  }
+
+  remove(toRemove: Type) {
+    this.all.splice(this.all.indexOf(toRemove));
+  }
+
+  removeIf(toRemoveFn: (value: Type) => boolean) {
+    const index = this.all.findIndex(toRemoveFn);
+    if (index === -1) {
+      return;
+    }
+    this.all.splice(index);
+  }
+
+  hasNextPage(): boolean {
+    return this.currentPageStart + this.pageSize < this.all.length;
```

Comment:
what happen if we have 7 videos and the page size is 5? Will it only show the first page of 5 videos and not the last 2 video?

Reply:
It will have one page of 5 and one page of 2.  I think that behavior is pretty typical for UC apps.

Reply:
nvm, misread the logic :D